File: livingrimoire start here/LivinGrimoire java/src/Auxiliary_Modules/EventChat.java
package Auxiliary_Modules;

import java.util.HashMap;
import java.util.Map;

public class EventChat {
    private final Map<String, UniqueResponder> dic;

    // Constructor
    public EventChat(UniqueResponder ur, String... args) {
        dic = new HashMap<>();
        for (String arg : args) {
            dic.put(arg, ur);
        }
    }

    // Add items
    public void addItems(UniqueResponder ur, String... args) {
        for (String arg : args) {
            dic.put(arg, ur);
        }
    }

    // Add key-value pair
    public void addKeyValue(String key, String value) {
        if (dic.containsKey(key)) {
            dic.get(key).addResponse(value);
        } else {
            dic.put(key, new UniqueResponder(value));
        }
    }
    // Get response
    public String response(String in1) {
        return dic.containsKey(in1) ? dic.get(in1).getAResponse() : "";
    }
}

File: livingrimoire start here/LivinGrimoire Kotlin/kotlin/auxiliary_modules/EventChat.kt
package auxiliary_modules

class EventChat(ur: UniqueResponder, vararg args: String) {
    private val dic: MutableMap<String, UniqueResponder>

    // Constructor
    init {
        dic = HashMap()
        for (arg in args) {
            dic[arg] = ur
        }
    }

    // Add items
    fun addItems(ur: UniqueResponder, vararg args: String) {
        for (arg in args) {
            dic[arg] = ur
        }
    }

    // Add key-value pair
    fun addKeyValue(key: String, value: String?) {
        if (dic.containsKey(key)) {
            dic[key]!!.addResponse(value!!)
        } else {
            dic[key] = UniqueResponder(value!!)
        }
    }

    // Get response
    fun response(in1: String): String {
        return if (dic.containsKey(in1)) dic[in1]!!.getAResponse() else ""
    }
}
